### ISSUE TITLE
Is there a way to checkout a specific ref of a subrepo?

### ISSUE THREAD

**Issue by nathanielc** [ingydotnet/git-subrepo#194]
Maybe something like

```
git subrepo pull <subdir> -r v0.1.0
```

**Comment by varac** [comment]
Yeah we're just asking ourselves the same question, would be good to know.

**Comment by perlpunk** [comment]
As far as I an see you can just use `--branch` for that. It accepts tags names.

**Comment by nathanielc** [comment]
It looks like `--branch` works to perform an initial clone but then later during a pull I get an error about not finding a common ancestor

```
$ git subrepo clone http://example.com/project.git deps/project -b v0.1
$ git subrepo pull deps/project -b master
git-subrepo: Can't find a common ancestor between refs/subrepo/deps/project/fetch and subrepo/deps/project. This might be the case if someone used push --force on the subrepo. You need to find out why and possibly reclone the subrepo.
```

I have even tried to clone from HEAD^ (using an explicit tag) and then pull master to update by a single commit on the same branch but I always get the common ancestor error.

**Comment by perlpunk** [comment]
@nathanielc this workflow also won't work with branches. Switching between branches for a subrepo is something you currently would do by simply recloning. See also the discussion in #184

**Comment by grimmySwe** [comment]
You can perform a clone with --force to reset your subdir to the exact version you specified.
```
$ git subrepo clone http://example.com/project.git deps/project -b v0.1
$ git subrepo clone --force http://example.com/project.git deps/project -b master
```

**Comment by nathanielc** [comment]
@grimmySwe Awesome, thanks for the tip. That will probably work for my needs.

**Comment by grimmySwe** [comment]
Sounds good, closing this for now.

**Comment by johnb003** [comment]
I feel like this isn't resolved.

The use case is if before using git-subrepo you made changes to another repository and want to apply your local changes at a point in the past, and pull forward.

In such a case you'd want to init the repository at a specific version, or clone it at a specific version, and then copy in your local changes, and commit that, before pulling down HEAD.

**Comment by grimmySwe** [comment]
@johnb003 I have tested some similar things on issue/328 branch (#328), so if you have some spare time I would be valuable to see if that does what you want. Note though that there are some changes there that uses functionality not available on 0.4.0 so you might want to test it out on a new branch in your repo.
